Source/Private/ASLayoutTransition.mm

```diff
@@ -156,18 +180,40 @@ - (void)calculateSubnodeOperationsIfNeeded
   ASLayout *pendingLayout = _pendingLayout->layout;
 
   if (previousLayout) {
+#if AS_IG_LIST_KIT
+    // IGListDiff completes in linear time O(m+n), so use it if we have it:
+    IGListIndexSetResult *result = IGListDiff(previousLayout.sublayouts, pendingLayout.sublayouts, IGListDiffEquality);
+    _insertedSubnodePositions = findNodesInLayoutAtIndexes(pendingLayout, result.inserts, &_insertedSubnodes);
+    _removedSubnodePositions = findNodesInLayoutAtIndexes(previousLayout, result.deletes, &_removedSubnodes);
+    for (IGListMoveIndex *move in result.moves) {
+      unowned ASDisplayNode *subnode = previousLayout.sublayouts[move.from].layoutElement;
+      _subnodeMoves.push_back(std::pair<ASDisplayNode *, NSUInteger>(subnode, move.to));
+    }
+
+    // Sort by ascending order of move destinations, this will allow easy loop of `insertSubnode:AtIndex` later.
+    std::sort(_subnodeMoves.begin(), _subnodeMoves.end(), [](std::pair<id<ASLayoutElement>, NSUInteger> a,
+            std::pair<ASDisplayNode *, NSUInteger> b) {
+      return a.second < b.second;
+    });
```

Comment:
Use `auto a, auto b` to save space

Reply:
"Generic lambdas are not supported by the compiler"

Reply:
What? That's crazy, it works in a sample project! OK must be something contextual, no worries

Reply:
that's interesting, do you guys ever build from the .xcworkspace/xcodeproj? I don't know much about BUCK but it seems like I have seen some other discrepancies